CHECKPOINTING.md

```diff
@@ -89,12 +98,35 @@ IoTHub().source(options)
     .run()
 ```
 
+# Post-Execution At-Least-Once Checkpointing
+
+At-least-once delivery semantic (ALOS) guarantees require that stream offsets are checkpointed after, and
+only after, the processing task is complete. This in turn requires a different approach to building Akka
+streaming graphs. Note that all configuration options listed above apply to the system, except for those
+governing frequency of saves (frequency, countThreshold, timeThreshold).
+
+The following example illustrates how code with ALOS looks like:
+
+```scala
+val options = SourceOptions().fromSavedOffsets()
+val hub = IoTHub()
+hub.source(options)
+    .map(m ⇒ jsonParser.readValue(m.contentAsString, classOf[Temperature]))
+    .filter(_.value > 100)
+    .via(console)
+    .to(hub.offsetSink(32)) //32 is the parallelism used to make concurrent saves
```

Comment:
does this snippet compile? messages going to `hub.offsetSink(32)` should be of type `MessageFromDevice`, but given the call to `map()` wouldn't this send `Temperature` objects to the sink?